addons/binding/org.openhab.binding.heos/README.md

```diff
@@ -0,0 +1,427 @@
+Bridge heos:bridge:main "name" [ipAddress="192.168.0.1", name="Default", unserName="xxx", password="123456"]  
+````
+
+### Player Configuration
+
+Players can be added via the PaperUI. All fields are then filled automatically.
+For manual configuration a player is defined as followed:
+
+````
+Thing heos:player:pid "name" [pid="123456789", name="name", model="modelName", ipAddress="192.168.0.xxx", type="Player"] 
+````
+
+PID behind the heos:player:--- should be changed as required. Every name or value can be used. It is recommended to use the player PID. Within the configuration the PID as well as the type field is mandatory. The rest is not required.
+If the PID isn't known it can be discovered by establishing a TelNet connection (port 1255) to one player and search for available players
+(Command: heos://player/get_players)
+within the network. For further details refer to the [HEOS CLI](http://rn.dmglobal.com/euheos/HEOS_CLI_ProtocolSpecification.pdf) specification.
+
+
+*If player is configured by PaperUI the UID of the player is equal to the player ID (PID) from the HEOS system*
+
+### Group Configuration
+
+```
+Thing heos:group:memberHash "name" [gid="123456789", name="name", model="modelName", ipAddress="192.168.0.xxx", type="Group", memberHash="123456789"] 
+```
+
+*If group is configured by PaperUI the group UID is calculated by the Hash value of the group members*
+
+Required fields are the GID which is the PID of the group leading player and the type which is Group here.  
+
+### Defining Bridge and Players together
+
+Defining Player and Bridge together. To ensure that the players and groups are attached to the bridge the definition can be like:
+
+```
+Bridge heos:bridge:main "Bridge" [ipAddress="192.168.0.1", name="Bridge", userName="userName", password="123456"] {
+	
+	player Kitchen "Kitchen"[pid="434523813", name="Kitchen", type="Player"]
+	player LivingRoom "Living Room"[pid="918797451", name="Living Room", type="Player"]
+  	player 813793755 "Bath Room"[pid="813793755", name="Bath Room", type="Player"]
+	
+}
+```
+
+## Channels
+
+Note:
+The channels have different paths if you configure our Things manual or via an UI. It is recommended to check the correct path via an UI.
+
+
+### Player provide the following channels:
+
+Channel ID | Item Type | Description
+----------------|-----------|-------------
+Control | Player | Play (also ON) / Pause (also OFF) / Next / Previous
+Volume | Dimmer | Volume control / also accepts "DECREASE" & "INCREASE"
+Mute | Switch | Mute the Player
+Title | String | Song Title
+Interpret | String | Song Interpret
+Album | String  | Album Title
+ImageUrl | String |The URL where the cover can be found 
+Inputs | String | The input to be switched to. Input values from HEOS protocol
+CurrentPosition | String | Shows the current trakc position in milliseconds
+Duration | String | The overall track duration in milliseconds
+Type | String | The type of the played media. Station or song for example
```

Comment:
If I switch to aux it keeps showing `station` (instead) as the type, however it does update station, input source and song:

![image](https://user-images.githubusercontent.com/2625279/43157452-5f88fec2-8f7d-11e8-9d15-98e525709b2d.png)

Reply:
And when switching to a song via my nas after that the input source does not update:

![image](https://user-images.githubusercontent.com/2625279/43157633-e2d1b5ee-8f7d-11e8-9173-e25af73168aa.png)

Reply:
Maybe it would make more sense to empty fields instead of filling them with a placeholder such as `No station`

Reply:
Unfortunately HEOS also indicates the `Aux In` as a station. But I will see what I can indicated instead.
Instead of `No Station` I will empty the field. That shouldn't be a problem.
The input source is not describe correctly. I have to improve that. It only indicates the external source. So not if played from local file or NAS. I wil check how it can be improved. Maybe by adding an additional channel for the general input source.

Reply:
I changed the behavior of that. Can you please check if it works better now?

Reply:
Did you happen to release it somewhere otherwise I can of course check out your branch :-)

Reply:
I released the latest version here as a quick draft.

https://github.com/Wire82/openhab2-addons/releases/tag/v2.4.0

Reply:
I literally have it running with a self compiled one for 5 minutes 👍

Reply:
Oh then I was a little bit to late. 😄